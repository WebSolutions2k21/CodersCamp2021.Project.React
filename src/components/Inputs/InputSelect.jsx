import { useState, useEffect } from 'react';
import { InputLabel, MenuItem, FormControl, Select } from '@mui/material';

import { useStyles } from './InputStyle';

export const InputSelect = ({ label, myNames = [], value, setValue }) => {
  const classes = useStyles();
  const [params, setParams] = useState([]);

  useEffect(() => {
    setParams(myNames);
  }, [myNames]);

  const handleChange = ({ target: { value } }) => setValue && setValue(value);

  return (
    <FormControl variant="standard" className={classes.int}>
      <InputLabel sx={{ ml: '17px', mr: '17px' }}>{label}</InputLabel>

      <Select
        disableUnderline
        value={value}
        onChange={handleChange}
        label={label}
        sx={{
          ml: '17px',
          mr: '17px',
          svg: { color: '#16BAC6' },
        }}
      >
        <MenuItem disabled value="">
          <em>{label}</em>
        </MenuItem>

        {params.map((name) => (
          <MenuItem key={name} value={name}>
            {name}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};

// Use import and <InputSelect label="type" myNames={["dog", "cat"]}/>
