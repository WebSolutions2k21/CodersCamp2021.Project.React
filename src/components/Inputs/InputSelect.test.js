import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { InputSelect } from './InputSelect';

afterEach(() => {
  cleanup();
});

test('should render InputSelect component', () => {
  render(<InputSelect label="type" myNames={["dog", "cat"]}/>);
  const inputComponent = screen.getByTestId('input-2');
  expect(inputComponent).toBeInTheDocument();
  expect(inputComponent).not.toBeNull();
  expect(inputComponent).not.toBeDisabled();
});


test('matches snapshot', () => {
  const tree = renderer.create(<InputSelect label="type" myNames={["dog", "cat"]}/>).toJSON();
  expect(tree).toMatchSnapshot();
});