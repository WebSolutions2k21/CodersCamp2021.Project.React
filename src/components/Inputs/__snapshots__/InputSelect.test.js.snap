// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiFormControl-root makeStyles-int-4 css-1nrlq1o-MuiFormControl-root"
  data-testid="input-2"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-1tmpjp0-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
  >
    type
  </label>
  <div
    className="MuiInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-wuj8x5-MuiInputBase-root-MuiInput-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      className="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
      >
        â€‹
      </span>
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
